### PR TITLE
proposal is now based on the current browsing language

### LINKED ISSUE #294: US #38 - Edit article cannot change language

### PR DESCRIPTION
**Pull request by vinyll**
fix #294

### CODE REVIEW

File: cafebabel/tests/test_drafts.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def test_create_draft_should_generate_article(client, user, editor):
         'author': user.id
     }, follow_redirects=True)
     assert response.status_code == 200
+    assert '<span>en</span>' in response
```
Comment (davidbgk) [inline cafebabel/tests/test_drafts.py]: Still passing the language within the test? What about hitting `/es/article/draft/new/` to be sure it's not the default that is rendered?

File: cafebabel/tests/test_drafts.py [hunk 2]
```diff
@@ -123,18 +120,16 @@ def test_draft_editing_should_update_content(client, user, editor):
...
+    draft = Article.objects.create(**data, language='en')
     updated_data = data.copy()
-    updated_data['language'] = 'fr'
     response = client.post(f'/en/article/draft/{draft.id}/edit/',
                            data=updated_data, follow_redirects=True)
     assert response.status_code == 200
     updated_draft = Article.objects.get(id=draft.id)
     assert updated_draft.id == draft.id
-    assert updated_draft.language == 'fr'
+    assert updated_draft.language == 'en'
```
Comment (davidbgk) [inline cafebabel/tests/test_drafts.py]: I don't think that assertion is still pertinent.

File: cafebabel/tests/test_proposals.py [hunk 3]
```diff
@@ -35,7 +33,7 @@ def test_proposal_send_email(app, client):
         assert len(outbox) == 1
         assert outbox[0].subject == 'Article proposal: Topic'
         assert outbox[0].body == f'''
-Language: {app.config['LANGUAGES'][0][0]}
+Language: en
```
Comment (davidbgk) [inline cafebabel/tests/test_proposals.py]: Can you check with `/de/article/proposal/new/` to be sure it's not a fallback?

File: cafebabel/templates/articles/proposals/create.html [hunk 4]
```diff
@@ -57,15 +46,10 @@ <h3>Wanna be part of Cafébabel project? That’s the right place</h3>
 
     <button type=submit>Suggest this article</button>
   </form>
-  <a class="button proposal-email" href=m&#x61;ilto:{{ EDITORS_EMAIL_DEFAULT|urlencode|obfuscate_email|safe }}>Or send your proposal by email</a>
+  <a class="button proposal-email">Or send your proposal by email</a>
```
Comment (davidbgk) [inline cafebabel/templates/articles/proposals/create.html]: I would have kept ` href=m&#x61;ilto:{{ editor_email|urlencode|obfuscate_email|safe }}`

File: cafebabel/templates/articles/proposals/create.html [hunk 5]
```diff
@@ -57,15 +46,10 @@ <h3>Wanna be part of Cafébabel project? That’s the right place</h3>
...
   <script>
-     const editorEmails = {{ editor_emails|to_json }}
-     const link = document.querySelector('.proposal-email')
-     const unobfuscateEmail = (email) => email.replace(/&#46;/g, '.').replace(/&#x61;/g, 'a')
-     document.querySelector('#field-language').addEventListener('change', (event) => {
-         const selectedLanguage = event.target.value
-         link.href = `mailto:${unobfuscateEmail(editorEmails[selectedLanguage])}`
-     })
+    const editorEmail = '{{ editor_email }}'
+    document.querySelector('.proposal-email').href = `mailto:${editorEmail.replace(/&amp;#46;/g, '.').replace(/&amp;#x61;/g, 'a')}`
```
Comment (davidbgk) [inline cafebabel/templates/articles/proposals/create.html]: But remove the JS altogether.

**Review by davidbgk — APPROVED** [review]
r+wc

File: cafebabel/tests/test_drafts.py (hunk 2 above)
Comment (vinyll) [reply cafebabel/tests/test_drafts.py]: this whole test is useless as long as we don't actually modify the data that we pass…

**Comment by davidbgk** [comment]
⚠️  I guess it will conflict with https://github.com/cafebabel/cafebabel.com/pull/305

File: cafebabel/templates/articles/edit.html [hunk 6]
```diff
@@ -18,17 +18,6 @@ <h1 class="edit {% if article.id %}active{% endif %}">
     <input id=title name=title value="{{ article.title or '' }}" required>
   </h1>
   <div class=article-settings>
-    <div class=select>
```
Comment (davidbgk) [inline cafebabel/templates/articles/edit.html:26]: ⚠️ articles-tags.js depends on that select, cf. https://github.com/cafebabel/cafebabel.com/pull/305#issuecomment-372165949 to kill one bug with two PRs ;)

Comment (vinyll) [reply cafebabel/templates/articles/edit.html:26]: Not sure what to do with this.
Suggestion @davidbgk?

File: cafebabel/static/css/styles.css [hunk 7]
```diff
@@ -1203,10 +1203,7 @@ select#author {
 }
 
 @media screen and (min-width: 750px) {
-  .create-article-page .select {
-    display: inline-block;
-    width: calc(50% - 0.3rem); /* space between the 2 .select */
-  }
+<<<<<<< HEAD
```
Comment (davidbgk) [inline cafebabel/static/css/styles.css]: 💣

File: cafebabel/templates/articles/proposals/create.html [hunk 8]
```diff
@@ -57,15 +46,10 @@ <h3>Wanna be part of Cafébabel project? That’s the right place</h3>
 
     <button type=submit>Suggest this article</button>
   </form>
-  <a class="button proposal-email" href=m&#x61;ilto:{{ EDITORS_EMAIL_DEFAULT|urlencode|obfuscate_email|safe }}>Or send your proposal by email</a>
+  <a class="button proposal-email" href="mailto:{{ editor_email }}">Or send your proposal by email</a>
```
Comment (davidbgk) [inline cafebabel/templates/articles/proposals/create.html]: If you remove the previous filters, the email is not anymore obfuscated?

File: cafebabel/articles/proposals/views.py [hunk 9]
```diff
@@ -23,26 +23,21 @@
...
 
-    editor_emails = {
-        lang: obfuscate_email(email)
-        for lang, email in current_app.config['EDITOR_EMAILS'].items()}
-
     return render_template(
         'articles/proposals/create.html',
-        EDITORS_EMAIL_DEFAULT=current_app.config['EDITORS_EMAIL_DEFAULT'],
-        editor_emails=editor_emails
+        editor_email=obfuscate_email(editor_email)
```
Comment (davidbgk) [inline cafebabel/articles/proposals/views.py:42]: Ah! Got it :)

File: cafebabel/articles/drafts/views.py [hunk 10]
```diff
@@ -1,6 +1,6 @@
 from flask import Blueprint, flash, redirect, render_template, request
 
-from ...core.helpers import editor_required
+from ...core.helpers import editor_required, current_language
```
Comment (davidbgk) [inline cafebabel/articles/drafts/views.py]: Useless import?

**Review by davidbgk — APPROVED** [review]
r+wc

**Pull request merged**